rasa/core/featurizers.py

```diff
@@ -133,6 +137,76 @@ def create_encoded_all_actions(self, domain: Domain) -> np.ndarray:
         return np.eye(domain.num_actions)
 
 
+class BOWSingleStateFeaturizer(CountVectorsFeaturizer, SingleStateFeaturizer):
+    def prepare_training_data_and_train(self, trackers_as_states, delimiter="_"):
+        """
+        Trains the vertorizers from real data;
+        Preliminary for when the featurizer is to be used for raw text; 
+        Args:
+             - trackers_as_states: real data as a dictionary
+             - delimiter: symbol to be used to divide into words
+        """
+        training_data = []
+        for tracker in trackers_as_states:
+            for state in tracker:
+                if state:
+                    state_keys = list(state.keys())
+                    state_keys = [
+                        Message(key.replace(delimiter, " ") + " CLS")
+                        for key in state_keys
+                    ]
+                    training_data += state_keys
+
+        training_data = TrainingData(training_examples=training_data)
+        self.train(training_data)
+
+    def prepare_from_domain(self, domain: Domain, delimiter="_") -> None:
+        """
+        Train the fecturizers based on the inputs gotten from the domain;
+        Args:
+            - domain: domain file
+        """
+
+        training_data = domain.input_states
+        training_data = [
+            Message(key.replace(delimiter, " ") + " CLS") for key in training_data
+        ]
+        training_data = TrainingData(training_examples=training_data)
+        self.train(training_data)
+
+    def encode(self, state: Dict[Text, float], delimiter="_", type_output="numpyarray"):
+        """
+        Encode the state into a numpy array or a sparse sklearn
+
+        Args:
+            - state: dictionary describing current state
+            - type output: type to return the features as (numpyarray or sklearn coo_matrix)
+        Returns:
+            - nparray(vocab_size,) or coo_matrix(1, vocab_size) 
+        """
+
+        if state is None:
+            return (
+                np.ones(len(self.vectorizers["text"].vocabulary_), dtype=np.int32) * -1
+            )
+
+        state_keys = [key.replace(delimiter, " ") for key in list(state.keys())]
+        attribute = "text"
+        message = Message(" ".join(state_keys))
+        message_tokens = self._get_processed_message_tokens_by_attribute(
+            message, attribute
+        )
+
+        message_tokens = " ".join(message_tokens)
```

Comment:
there is no need to join message_tokens, the last output (for `CLS` token) will contain bag of word features